multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -287,3 +322,67 @@ func (r *MemberClusterAnnounceReconciler) GetMemberClusterStatuses() []multiclus
 
 	return status
 }
+
+func (r *MemberClusterAnnounceReconciler) addMemberToClusterSet(memberClusterAnnounce *multiclusterv1alpha1.MemberClusterAnnounce) error {
+	clusterSetID := memberClusterAnnounce.ClusterSetID
+	clusterSet := &multiclusterv1alpha1.ClusterSet{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Namespace: memberClusterAnnounce.Namespace, Name: clusterSetID}, clusterSet); err != nil {
+		if errors.IsNotFound(err) {
+			klog.ErrorS(err, "ClusterSet not found in leader cluster", "ClusterSet", clusterSetID)
+			return err
+		}
+		klog.ErrorS(err, "Failed to get ClusterSet in leader cluster", "ClusterSet", clusterSetID)
+		return err
+	}
+
+	exist := false
+	for _, member := range clusterSet.Spec.Members {
+		if member.ClusterID == memberClusterAnnounce.ClusterID {
+			exist = true
+			break
+		}
+	}
+
+	if exist {
+		return fmt.Errorf("member cluster %s already exists in ClusterSet %s", memberClusterAnnounce.ClusterID, clusterSetID)
```

Comment:
should this error be ignored?

Reply:
Is this resolved? I feel this means there are cluster ID conflicts in a ClusterSet, should it be in webhook to make sure user know what's going on, so they can take right action?